### PR TITLE
[api][frontend][BCCIS-#121] Front end change for CRC field 

### PR DESCRIPTION
**Pull request by fwpushan**
-----
1. Add new date field
2. Update interface
3. Make existing legacy field optional if non date object present.
---

### CODE REVIEW

File: api/src/interpreter/dto/create-interpreter.dto.ts [hunk 1]
```diff
@@ -72,7 +73,7 @@ export class CreateInterpreterDto {
   })
   @IsOptional()
   @ValidateIf(e => e.postal !== '')
-  @IsPostalCode('CA')
+  // @IsPostalCode('CA')
```
Comment (will-freshworks) [inline api/src/interpreter/dto/create-interpreter.dto.ts:76]: Why has this been removed, along with the phone number check?

File: frontend/src/components/form/inputs/Interpreter.tsx [hunk 2]
```diff
@@ -1,28 +1,46 @@
...
 
 type GridItemInputProps = {
   name: string;
   label: string;
   rows?: any;
-}
+  initialValue?: any;
+};
+
+const setDateFormat = (date?: Date): string => {
```
Comment (will-freshworks) [inline frontend/src/components/form/inputs/Interpreter.tsx:29]: Not sure this should begin with `set`, makes me think it's a property setter. Possibly `formatDate(...)`?

File: frontend/src/components/form/inputs/Interpreter.tsx [hunk 3]
```diff
@@ -42,89 +60,182 @@ const StyledField = ({ name, label, rows = { xs: 6 } }: GridItemInputProps) => (
...
+);
+
+const StyledDateField = ({
+  name,
+  label,
+  rows = { xs: 6 },
+  initialValue,
+}: GridItemInputProps) => {
+  const [date, setDate] = React.useState('');
+  const [, , setField] = useField(name);
```
Comment (will-freshworks) [inline frontend/src/components/form/inputs/Interpreter.tsx:72]: Don't like the name `setField` here, possibly you could pattern match it out like:
```
const [, , { setValue }] = useField(name);
```

**Review by will-freshworks — CHANGES_REQUESTED** [review]
Minor comments. @michel-fresh could add much better ones I'm sure.

File: frontend/src/components/form/inputs/Interpreter.tsx (hunk 2 above)
Comment (fwpushan) [reply frontend/src/components/form/inputs/Interpreter.tsx]: FIXED

**Review by will-freshworks — APPROVED** [review]

**Pull request merged**